File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/TipCalculator.kt
package com.qaantum.calculatorhub.calculators

data class TipResult(
    val tipAmount: Double,
    val totalBill: Double,
    val amountPerPerson: Double
)

class TipCalculator {
    fun calculate(
        billAmount: Double,
        tipPercentage: Double,
        splitCount: Int = 1
    ): TipResult {
        val tip = billAmount * (tipPercentage / 100.0)
        val total = billAmount + tip
        val perPerson = total / splitCount
        
        return TipResult(
            tipAmount = tip,
            totalBill = total,
            amountPerPerson = perPerson
        )
    }
}

File: Calculator/ios-native/CalculatorHub/Calculators/TipCalculator.swift
import Foundation

struct TipResult {
    let tipAmount: Double
    let totalBill: Double
    let amountPerPerson: Double
}

class TipCalculator {
    func calculate(billAmount: Double, tipPercentage: Double, splitCount: Int = 1) -> TipResult {
        let tip = billAmount * (tipPercentage / 100.0)
        let total = billAmount + tip
        let perPerson = total / Double(splitCount)
        
        return TipResult(
            tipAmount: tip,
            totalBill: total,
            amountPerPerson: perPerson
        )
    }
}
